src/components/FeatureViews/Registration/RegistrationCart/RegistrationCartContainer.js

```diff
@@ -87,21 +86,36 @@ export default function RegistrationCartContainer() {
 	const studentIds = (Object.keys(registrationCartState).length > 0 && Object.keys(registrationCartState)) ||
 		currentParent.studentList;
 	// create list of courses to fetch
-	const courseIds = [].concat.apply([], Object.values(registrationCartState))
-		.map(({course}) => course.id);
-	const {data, loading} = useQuery(GET_STUDENT_INFOS, {variables: {userIds: studentIds}});
-	const coursesResponse = useQuery(GET_COURSES_TO_REGISTER, {variables: {courseIds: courseIds}});
+	const courseIds = Object.values(registrationCartState).filter(reg => reg).length > 0 &&
+		[].concat.apply([], Object.values(registrationCartState))
+			.filter(registration => registration)
+			.map(({course}) => course.id);
+
+	const {data, loading, error} = useQuery(GET_COURSES_AND_STUDENTS_TO_REGISTER,
+		{
+			variables: {userIds: studentIds, courseIds: courseIds},
+			skip: !courseIds,
+		});
 	const [createRegistrationCart, createRegistrationCartResponse] = useMutation(CREATE_REGISTRATION_CART, {
-		variables: {parent: currentParent.user.id},
+		variables: {parent: currentParent?.user.id},
 		onCompleted: () => {
 			setParentConfirmation(true);
 		},
 		onError: (error) => console.error(error.message),
 	});
 
 	useEffect(() => {
-		if (!coursesResponse.loading && Object.values(registrationCartState).length > 0) {
-			const courseData = coursesResponse.data.courses;
+		dispatch({
+			type: types.INIT_COURSE_REGISTRATION,
+			payload: {},
+		})
+	}, []);
```

Comment:
Not sure, but do we add dependency's into these useEffect hooks? 

like ```[types.INIT_COURSE_REGISTRATION]```, correct me if I'm wrong

Reply:
yeah, we do. i just wanted to confirm that this would run on page load. nothing should change if i put it in though. good catch